bitacross-worker/core-primitives/stf-state-handler/src/state_handler.rs

```diff
@@ -236,6 +236,21 @@ where
 		let (state, _) = self.load_cloned(&old_shard)?;
 		self.reset(state, &new_shard)
 	}
+
+	fn force_migrate_shard(&self, new_shard: ShardIdentifier) -> Result<Self::HashType> {
+		if self.shard_exists(&new_shard)? {
+			let (_, state_hash) = self.load_cloned(&new_shard)?;
+			return Ok(state_hash)
+		}
+		let old_shard = match self.list_shards()? {
+			shards if shards.len() == 1 => shards[0],
```

Comment:
If it's the same shard (the worker was just restarted or code change was not inside enclave-runtime) does it still make sense to migrate it ? I think we should skip shard migration in such case.

Reply:
It does skip it - see a few lines above.

If we fall into this branch it means it has only 1 shard, but a different shard to the current one

Reply:
True true, I was misleaded by log - it keeps on saying migration was succesfull.

Reply:
Ah it's also a good point - so log saying "migration successful" can actually mean "migration was skipped because the shard matches the current mrenclave"

Reply:
It can be improved later though - either we introduce additional `MigrationResult` enum, or simply add a log saying migration was skipped